runtime/tests/integration_tests.rs

```diff
@@ -425,6 +430,148 @@ fn test_invoke_dpe_get_certificate_chain_cmd() {
+        .copy_from_slice(&rotate_ctx_resp_buf);
+
+    assert!(caliptra_common::checksum::verify_checksum(
+        rotate_ctx_resp_hdr.hdr.chksum,
+        0x0,
+        &rotate_ctx_resp_buf[core::mem::size_of_val(&rotate_ctx_resp_hdr.hdr.chksum)..],
+    ));
+
+    let rotate_ctx_resp = NewHandleResp::read_from(
+        &rotate_ctx_resp_hdr.data[..rotate_ctx_resp_hdr.data_size as usize],
+    )
+    .unwrap();
+    let mut handle = rotate_ctx_resp.handle;
+
+    // Call DeriveChild with PL0 enough times to breach the threshold on the last iteration.
+    // Note that this loop runs exactly PL0_DPE_ACTIVE_CONTEXT_THRESHOLD times, but due to auto
+    // initialization of DPE, we have one extra context, so the last iteration of this loop
+    // is expected to throw a threshold breached error.
+    for i in 0..InvokeDpeCmd::PL0_DPE_ACTIVE_CONTEXT_THRESHOLD {
+        let mut cmd_data: [u8; 512] = [0u8; InvokeDpeReq::DATA_MAX_SIZE];
+        let derive_child_cmd = DeriveChildCmd {
+            handle,
+            data: [0u8; DPE_PROFILE.get_hash_size()],
+            flags: DeriveChildFlags::RETAIN_PARENT,
+            tci_type: 0,
+            target_locality: 0,
+        };
+        let derive_child_cmd_hdr = CommandHdr::new_for_test(Command::DERIVE_CHILD);
+        let derive_child_cmd_hdr_buf = derive_child_cmd_hdr.as_bytes();
+        cmd_data[..derive_child_cmd_hdr_buf.len()].copy_from_slice(derive_child_cmd_hdr_buf);
+        let derive_child_cmd_buf = derive_child_cmd.as_bytes();
+        cmd_data[derive_child_cmd_hdr_buf.len()
+            ..derive_child_cmd_hdr_buf.len() + derive_child_cmd_buf.len()]
+            .copy_from_slice(derive_child_cmd_buf);
+        let derive_child_mbox_cmd = InvokeDpeReq {
+            hdr: MailboxReqHeader { chksum: 0 },
+            data: cmd_data,
+            data_size: (derive_child_cmd_hdr_buf.len() + derive_child_cmd_buf.len()) as u32,
+        };
+
+        let checksum = caliptra_common::checksum::calc_checksum(
+            u32::from(CommandId::INVOKE_DPE),
+            &derive_child_mbox_cmd.as_bytes()[4..],
+        );
+
+        let derive_child_mbox_cmd = InvokeDpeReq {
+            hdr: MailboxReqHeader { chksum: checksum },
+            ..derive_child_mbox_cmd
+        };
+
+        // If we are on the last call to DeriveChild, expect that we get a PL0_ACTIVE_DPE_CONTEXT_THRESHOLD_EXCEEDED error.
+        if i == InvokeDpeCmd::PL0_DPE_ACTIVE_CONTEXT_THRESHOLD - 1 {
+            let resp = model
+                .mailbox_execute(
+                    u32::from(CommandId::INVOKE_DPE),
+                    derive_child_mbox_cmd.as_bytes(),
+                )
+                .unwrap_err();
+            if let ModelError::MailboxCmdFailed(code) = resp {
+                assert_eq!(
+                    code,
+                    caliptra_drivers::CaliptraError::RUNTIME_PL0_ACTIVE_DPE_CONTEXT_THRESHOLD_EXCEEDED.into()
+                );
+            }
+            break;
+        }
```

Comment:
nit: If you update the loop to be `for i in 0..InvokeDpeCmd::PL0_DPE_ACTIVE_CONTEXT_THRESHOLD`, you can add this check outside the loop and ditch the `if`.

Reply:
But then derive_child_mbox_cmd would be out of scope, right? I agree this isn't the cleanest way to do it, but it would be a lot of code repetition to move this to outside of the loop I think.